src/core/src/core_logic/MaintenanceWindow.py

```diff
@@ -59,9 +59,13 @@ def get_remaining_time_in_minutes(self, current_time=None, log_to_stdout=False):
 
         return remaining_time_in_minutes
 
-    def is_package_install_time_available(self, remaining_time_in_minutes=None):
+    def is_packages_install_time_available(self, remaining_time_in_minutes=None, number_of_packages = 1, reboot_manager=None):
         """Check if time still available for package installation"""
-        cutoff_time_in_minutes = Constants.REBOOT_BUFFER_IN_MINUTES + Constants.PACKAGE_INSTALL_EXPECTED_MAX_TIME_IN_MINUTES
+        cutoff_time_in_minutes = Constants.PACKAGE_INSTALL_EXPECTED_MAX_TIME_IN_MINUTES * number_of_packages
```

Comment:
At the boundary of the maintenance window, are you going to back off on the number of packages?

For example, say the number is 10 and time for one is 1 minute. Ignoring reboot time, if you have 5-6 minutes left, will you back off until you know you can do 5 instead of missing all 10?

Reply:
I just tested and found that this scenario is not handled as per what I have explained above. The variable maintenance_window_exceeded is set to True in line 404. Ideally this should not be set there. I think it is better to add log there and do not set maintenance_window_exceeded to True. The variable maintenance_window_exceeded should be set to True only during sequential patching if the condition is met. I will handle this case.

Reply:
Handled this scenario and manually tested the same. Now if there is not enough time to install patches in batches then the remaining packages are installed sequentially as per existing way of installing packages.
Commit in which this scenario is handled: https://github.com/Azure/LinuxPatchExtension/pull/171/commits/7d15edced21bd9352f080803ffe8a65fd8e4a073

Reply:
""The variable maintenance_window_exceeded is set to True in line 404.""

Haven't gone through the remaining changes here, but can add a reference of the file/code line for this ^?

maintenance_window_exceeded should also be considered during individual package install. In the existing scenario (one package install at a time), in case the extension runs out of maintenance_window, patching is stopped

Reply:
@GAURAVRAMRAKHYANI I don't think this is addressed yet.

Reply:
@rane-rajasi I could not understand this. Could you please explain?

Reply:
@rane-rajasi I could not understand the comment. Could you please explain?

Reply:
@GAURAVRAMRAKHYANI, in your earlier comment, you mentioned "The variable maintenance_window_exceeded is set to True in line 404". 

In which file is this set to true? Since MaintenanceWindow.py isn't that long.

The other part of this comment was to ensure, maintenance_window_exceeded check is still in use during single/individual package install.

Reply:
Got it. I mentioned only line number but missed to add reference to file.
After new iterations, the reference have changed.
The file is PatchInstaller.py.

Using variable maintenance_window_batch_cutoff_reached in batch patching. This variable is true when there is not enough time to install packages in batches. If this variable is true, then logging in the method install_updates and install rest of the packages sequentially. Using the variable maintenance_window_exceeded during sequential installation.